scripts/boot-master/functions.sh

```diff
@@ -38,7 +38,7 @@ function parse_icpversion() {
   then
     # We should autodetect the version if loaded from tarball
     # For now we grab the first matching image in docker image list.
-    read -r repo tag <<<$(docker image list | awk -v pat=$DefaultRepo ' $1 ~ pat { print $1 " " $2 ; exit }')
+    read -r repo tag <<<$(sudo docker image list | awk -v pat=$DefaultRepo ' $1 ~ pat { print $1 " " $2 ; exit }')
```

Comment:
These `sudo` commands are needed to ensure that the commands never fail in case the docker port is not accessible to the current user. We never add `ssh_user` to the docker group if we install it. Should we do that?

Reply:
Adding the user to the docker group should be dealt with by 
https://github.com/ibm-cloud-architecture/terraform-module-icp-deploy/blob/master/scripts/boot-master/install-docker.sh#L151 for Ubuntu or
https://github.com/ibm-cloud-architecture/terraform-module-icp-deploy/blob/master/scripts/boot-master/install-docker.sh#L165 for RHEL which essentially calls this script:
https://github.com/ibm-cloud-architecture/terraform-module-icp-deploy/blob/master/scripts/common/docker-user.sh
This has always been sitting in a separated `null_resource` to ensure that dependent resources would be called after this, essentially ensuring fresh login of the `ssh_user` after `usermod -a -G` was called.
`icp-docker` runs pretty early on the boot node, but there may have been some changes in the way docker or Terraform works, or in the order of the resources are called that has rendered this approach inoperable.